code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -131,15 +134,53 @@ butcher
 /datum/aiTask/succeedable/replicate/on_reset()
 	has_started = 0
 
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////
+// NEST + REPLICATION GOAL
+// targets: valid nesting sites
+// precondition: 120 resources, no flocktiles in view
+/datum/aiTask/sequence/goalbased/nest
+	name = "nesting"
+	weight = 6
+	can_be_adjacent_to_target = 1
+	max_dist = 2
+
+/datum/aiTask/sequence/goalbased/nest/New(parentHolder, transTask)
+	..(parentHolder, transTask)
+	add_task(holder.get_instance(/datum/aiTask/succeedable/build, list(holder)))
+
+/datum/aiTask/sequence/goalbased/nest/precondition()
+	. = 0
+	var/mob/living/critter/flock/drone/F = holder.owner
+	if(F?.can_afford(120))
+		. = 1 //we can afford
+		for(var/turf/simulated/floor/feather/T in view(max_dist, holder.owner))
+			return FALSE //but there's a flocktile in view
```

Comment:
So they.. can't lay eggs if they're near the flock?

Reply:
it can't do the _nesting_ behaviour if it's near the flock, it will just do the standard replication behaviour instead.